homeassistant/components/flume/coordinator.py

```diff
@@ -23,13 +34,89 @@ def __init__(self, hass: HomeAssistant, flume_device) -> None:
 
     async def _async_update_data(self) -> None:
         """Get the latest data from the Flume."""
-        _LOGGER.debug("Updating Flume data")
         try:
             await self.hass.async_add_executor_job(self.flume_device.update_force)
         except Exception as ex:
             raise UpdateFailed(f"Error communicating with flume API: {ex}") from ex
         _LOGGER.debug(
-            "Flume update details: values=%s query_payload=%s",
+            "Flume Device Data Update values=%s query_payload=%s",
             self.flume_device.values,
             self.flume_device.query_payload,
         )
+
+
+class FlumeDeviceConnectionUpdateCoordinator(DataUpdateCoordinator[None]):
+    """Date update coordinator to read connected status from Devices endpoint."""
+
+    def __init__(self, hass: HomeAssistant, flume_devices: FlumeDeviceList) -> None:
+        """Initialize the Coordinator."""
+        super().__init__(
+            hass,
+            name=DOMAIN,
+            logger=_LOGGER,
+            update_interval=DEVICE_CONNECTION_SCAN_INTERVAL,
+        )
+
+        self.flume_devices = flume_devices
+        self.connected: dict[str, bool] = {}
+
+    def _update_connectivity(self) -> None:
+        """Update device connectivity.."""
+        self.connected = {
+            device["id"]: device["connected"]
+            for device in self.flume_devices.get_devices()
+        }
+        _LOGGER.debug("Connectivity %s", self.connected)
+
+    async def _async_update_data(self) -> None:
+        """Update the device list."""
+        try:
+            await self.hass.async_add_executor_job(self._update_connectivity)
+        except Exception as ex:
```

Comment:
What exception do we expect? Can we narrow down the exception?

Reply:
I honestly have no idea ... I'll try to figure it out...